### PR TITLE
inference api symbol hidden

### LINKED ISSUE #12042: hide symbols in `paddle_inference_api.so`
something like this:
```
  3087: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  UND _ZTISt16invalid_argument@
  3092: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt9basic_iosIcSt11char
  3093: 000000000008ae00  1551 FUNC    GLOBAL DEFAULT   12 _ZN6google9SendEmailEPKcS
  3100: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSs6appendERKSs@@GLIBCX
  3104: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND _ZN6paddle9framework9make
  3109: 0000000000089290   163 FUNC    GLOBAL DEFAULT   12 _ZN6google16posix_strerro
  3114: 000000000008f420    12 FUNC    GLOBAL DEFAULT   12 _ZN6google24glog_internal
  3119: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND open@@GLIBC_2.2.5
  3129: 0000000000099c30     8 FUNC    GLOBAL DEFAULT   12 _ZN6google13VersionString
  3130: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  UND _ZTIl@@CXXABI_1.3
  3135: 00000000002e03c0     1 OBJECT  GLOBAL DEFAULT   26 _ZN3fLB15FLAGS_noversionE
  3152: 000000000008c300   228 FUNC    GLOBAL DEFAULT   12 _ZN6google26GetExistingTe
  3157: 00000000002cfd28     8 OBJECT  GLOBAL DEFAULT   26 _ZN6google14LogDestinatio
  3178: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND strtod@@GLIBC_2.2.5
  3182: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  UND stdout@@GLIBC_2.2.5
  3187: 000000000008b460   142 FUNC    GLOBAL DEFAULT   12 _ZN6googlelsERSoRKNS_15PR
  3195: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND time@@GLIBC_2.2.5
  3198: 000000000009bb20     5 FUNC    GLOBAL DEFAULT   12 _ZN6google21RegisterFlagV
  3200: 000000000005ec30     0 FUNC    GLOBAL DEFAULT    9 _init
  3203: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt15basic_streambufIcS
  3205: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fflush@@GLIBC_2.2.5
  3206: 00000000000a0e20    13 FUNC    GLOBAL DEFAULT   12 _ZN6google21ParseCommandL
  3213: 000000000008f4d0    47 FUNC    GLOBAL DEFAULT   12 _ZN6google24glog_internal
  3214: 00000000002cfc60     4 OBJECT  GLOBAL DEFAULT   25 _ZN3fLI28FLAGS_tab_comple
```

### PR DESCRIPTION
**Pull request by tensor-tang**
fix #12042

The result should be like this:

Before:
```
a280   777 FUNC    GLOBAL DEFAULT   12 _ZN6paddle21NativePaddleP
  1049: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fcntl@@GLIBC_2.2.5
  1050: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt9basic_iosIcSt11char
  1051: 00000000000295e0  1551 FUNC    GLOBAL DEFAULT   12 _ZN6google9SendEmailEPKcS
  1052: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSs6appendERKSs@@GLIBCX
  1054: 0000000000027a70   163 FUNC    GLOBAL DEFAULT   12 _ZN6google16posix_strerro
  1055: 000000000002db00    12 FUNC    GLOBAL DEFAULT   12 _ZN6google24glog_internal
  1056: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND open@@GLIBC_2.2.5
  1058: 0000000000038310     8 FUNC    GLOBAL DEFAULT   12 _ZN6google13VersionString
  1064: 000000000002aae0   228 FUNC    GLOBAL DEFAULT   12 _ZN6google26GetExistingTe
  1067: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND strtod@@GLIBC_2.2.5
  1069: 0000000000029c40   142 FUNC    GLOBAL DEFAULT   12 _ZN6googlelsERSoRKNS_15PR
  1070: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND time@@GLIBC_2.2.5
  1071: 000000000003a200     5 FUNC    GLOBAL DEFAULT   12 _ZN6google21RegisterFlagV
  1072: 0000000000012bf0     0 FUNC    GLOBAL DEFAULT    9 _init
  1074: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt15basic_streambufIcS
  1075: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fflush@@GLIBC_2.2.5
  1076: 000000000003f500    13 FUNC    GLOBAL DEFAULT   12 _ZN6google21ParseCommandL
  1077: 000000000002dbb0    47 FUNC    GLOBAL DEFAULT   12 _ZN6google24glog_internal
```

After
```
 3033: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt16invalid_argumentD1
  3035: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt15basic_streambufIcS
  3048: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT  UND _ZNK6paddle9framework6Ten
  3049: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND perror@@GLIBC_2.2.5
  3059: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSaIcEC1Ev@@GLIBCXX_3.4
  3065: 000000000004b4ec   110 FUNC    GLOBAL DEFAULT   12 _ZN6paddle9PaddleBufaSERK
  3067: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND __gxx_personality_v0@@CXX
  3072: 000000000004b406    72 FUNC    GLOBAL DEFAULT   12 _ZN6paddle15PaddleDtypeSi
  3073: 000000000004c238  2691 FUNC    GLOBAL DEFAULT   12 _ZN6paddle21NativePaddleP
  3074: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSt19basic_ostringstrea
  3076: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fprintf@@GLIBC_2.2.5
  3079: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSsC1EOSs@@GLIBCXX_3.4.
```

As you can see something like `1051: 00000000000295e0  1551 FUNC    GLOBAL DEFAULT   12 _ZN6google9SendEmailEPKcS` has been removed.

### CODE REVIEW

File: paddle/contrib/inference/paddle_inference_api.map [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+	global:
+		*paddle*;
```
Comment (tensor-tang) [inline paddle/contrib/inference/paddle_inference_api.map:3]: @Superjomn 

Use `*paddle*` or the explicit name we want to expose? Which one do we prefer?

Comment (Superjomn) [reply paddle/contrib/inference/paddle_inference_api.map:3]: `*paddle*` should be enough currently, we can change to the explicit names if the users need that.

Comment (tensor-tang) [reply paddle/contrib/inference/paddle_inference_api.map:3]: Cool

**Review by Superjomn — COMMENTED** [review]
可以加个 COMMAND 自动验证只包含 *paddle*，不包含 protobuf 等symbol

比如     


xxx.sh
```bash
lib_path=$1
num_paddle_symbols=$(nm $lib_path | grep "*paddle*" | wc -l)
[ $num_paddle_symbols -le 0] && exit -1

num_other_symbols=$(nm $lib_path | grep "*google*" | wc -l)
[ $num_other_symbols -gt 0] && exit -1
```
cmake:
```cmake
execute_process(COMMAND bash -c "./xxx.sh")
```

**Comment by tensor-tang** [comment]
这个想法是挺好，但是很难做到check所有symbol，只能是遇到哪个有问题了就check哪个，比如google的。

其他的第三方库也不知道程序引用了哪些函数，会暴露哪些出来。

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**